flows/engine/session.go

```diff
@@ -159,6 +166,9 @@ func (s *session) start(trigger flows.Trigger) (flows.Sprint, error) {
 		return sprint, err
 	}
 
+	// ensure dynamic groups are correct
+	s.ensureDynamicGroups(sprint.LogEvent)
```

Comment:
this was previously happening in resumes and triggers but easier to do it in the session, to ensure it happens after things that can affect the groups.. like setting input which changes last_seen_on